ext/rotoscope/extconf.rb

```diff
@@ -2,4 +2,6 @@
 require "mkmf"
 
 $CFLAGS << ' -std=c99 -Wall -Werror -Wno-declaration-after-statement'
+$defs << "-D_XOPEN_SOURCE=600 -D_POSIX_C_SOURCE=200112L"
```

Comment:
The requirements `_POSIX_C_SOURCE >= 1 || _XOPEN_SOURCE || _POSIX_SOURCE` mean that you don't need both of these macros, you just need one of them defined.  Since the X/Open standard is a superset of posix and for fileno we only need POSIX.1, you can just use `-D_POSIX_SOURCE` until we need features from the other standards.  Actually, `_POSIX_C_SOURCE` gets defined automatically by the standard headers when one of the others is defined.